satpy/tests/reader_tests/test_fci_l1c_nc.py

```diff
@@ -282,236 +292,220 @@ def _get_reader_with_filehandlers(filenames, reader_configs):
+                assert res[ch].dtype == np.uint16
+                assert res[ch].attrs["calibration"] == "counts"
+                assert res[ch].attrs["units"] == "count"
+                if ch == 'ir_38':
+                    numpy.testing.assert_array_equal(res[ch][~0], 1)
+                    numpy.testing.assert_array_equal(res[ch][0], 5000)
+                else:
+                    numpy.testing.assert_array_equal(res[ch], 1)
+
+    @pytest.mark.parametrize('filehandler,channels,filenames,expected_res_n', [
+        (FakeFCIFileHandlerFDHSI, _chans_fdhsi, _test_filenames['fdhsi'], 16),
+        (FakeFCIFileHandlerHRFI, _chans_hrfi, _test_filenames['hrfi'], 4),
+
+    ])
+    def test_load_radiance(self, reader_configs, filehandler, channels, filenames,
+                           expected_res_n):
         """Test loading with radiance."""
-        from satpy.tests.utils import make_dataid
-
-        filenames = [
-            "W_XX-EUMETSAT-Darmstadt,IMG+SAT,MTI1+FCI-1C-RRAD-FDHSI-FD--"
-            "CHK-BODY--L2P-NC4E_C_EUMT_20170410114434_GTT_DEV_"
-            "20170410113925_20170410113934_N__C_0070_0067.nc",
-        ]
-
-        reader = _get_reader_with_filehandlers(filenames, reader_configs)
-        res = reader.load(
-            [make_dataid(name=name, calibration="radiance") for name in
-             self._chans["solar"] + self._chans["terran"]], pad_data=False)
-        assert 16 == len(res)
-        for ch in self._chans["solar"] + self._chans["terran"]:
-            assert res[ch].shape == (200, 11136)
-            assert res[ch].dtype == np.float64
-            assert res[ch].attrs["calibration"] == "radiance"
-            assert res[ch].attrs["units"] == 'mW m-2 sr-1 (cm-1)-1'
-            assert res[ch].attrs["radiance_unit_conversion_coefficient"] == 1234.56
-            if ch == 'ir_38':
-                numpy.testing.assert_array_equal(res[ch][~0], 15)
-                numpy.testing.assert_array_equal(res[ch][0], 9700)
-            else:
-                numpy.testing.assert_array_equal(res[ch], 15)
-
-    def test_load_reflectance(self, reader_configs):
+        with mocked_basefilehandler(filehandler):
+            reader = _get_reader_with_filehandlers(filenames, reader_configs)
+            res = reader.load(
+                [make_dataid(name=name, calibration="radiance") for name in
+                 channels["solar"] + channels["terran"]], pad_data=False)
+            assert expected_res_n == len(res)
+            for ch in channels["solar"] + channels["terran"]:
+                assert res[ch].shape == (200, 11136)
+                assert res[ch].dtype == np.float64
+                assert res[ch].attrs["calibration"] == "radiance"
+                assert res[ch].attrs["units"] == 'mW m-2 sr-1 (cm-1)-1'
+                assert res[ch].attrs["radiance_unit_conversion_coefficient"] == 1234.56
+                if ch == 'ir_38':
+                    numpy.testing.assert_array_equal(res[ch][~0], 15)
```

Comment:
Why `~0`?  That's just `-1`?

Reply:
Maybe I thought ~0 is cooler than -1 :) No idea to be honest, it's been a while. Changed now.